legion/engine/core/defaults/defaultcomponents.hpp

```diff
@@ -186,9 +187,39 @@ namespace legion::core
     struct mesh_filter : public mesh_handle
     {
         mesh_filter() = default;
-        explicit mesh_filter(const mesh_handle& src) { id = src.id; };
+        explicit mesh_filter(const mesh_handle& src) : mesh_handle(src) {}
 
         bool operator==(const mesh_filter& other) const { return id == other.id; }
+
+        template<class Archive>
+        void save(Archive& oa)
+        {
+            bool debug = false;
+            if (id != invalid_id)
+                oa(id, cereal::make_nvp("Debug", debug), cereal::make_nvp("Filepath", get().second.filePath));
```

Comment:
I dont know what 'oa' function does, perhaps give it a more descriptive name

Reply:
the 'oa' is short for 'outarchive'. which is just the name for the object cereal uses to pass data around serialize functions. It's just shorthand for these functions. I can change it if you really want though.

Reply:
doesnt seem to be a big issue, I'll alow it